inc/commondbtm.class.php

```diff
@@ -271,20 +271,43 @@ public static function getFromIter(DBmysqlIterator $iter) {
     * @return boolean|array
     */
    public function getFromDBByCrit(array $crit) {
+      return $this->getFromDBByRequest(['WHERE' => $crit]);
+   }
+
+   /**
+    * Retrieve an item from the database by request. The request is an array
+    * similar to the one expected in DB::request().
+    *
+    * @since 9.3
+    *
+    * @see DB::request()
+    *
+    * @param array $request expression
+    *
+    * @return boolean true if succeed else false
+    **/
+   public function getFromDBByRequest(array $request) {
       global $DB;
 
-      $crit = ['SELECT' => 'id',
-               'FROM'   => $this->getTable(),
-               'WHERE'  => $crit];
+      // Limit the request to the useful expressions
+      $request = array_intersect_key($request, [
+         'WHERE' => '',
+         'LEFT JOIN' => '',
+         'INNER_JOIN' => '',
+         'START' => '',
+         'LIMIT' => '',
+      ]);
+      $request['FROM'] = $this->getTable();
+      $request['SELECT'] = $this->getTable() . '.id';
```

Comment:
This is maybe too restrictive (see my comment on `getFromDB` below.

Either drop the `SELECT` part  (or set it to `$this->getTable() . '*'`); or make it configurable as well?

Reply:
I get the point with the next comment.

As the method shall populate the instance ($instance->fields array) the select must be definitively ```table.*```